### PR TITLE
INDY-1289: support read-only state in read_ledger with RocksDB.

### PR DESCRIPTION
**Pull request by sergey-shilov**
Since read-only mode is supported for RocksDB a copy of
ledger is not created in case of RocksDB-based ledger.

Also added additional checks to avoid removing of the
ledger instead of it's copy.

Signed-off-by: Sergey Shilov <sergey.shilov@dsr-company.com>

### CODE REVIEW

File: scripts/read_ledger [hunk 1]
```diff
@@ -153,15 +156,26 @@ def make_copy_of_ledger(data_dir):
...
+    config = getConfig()
 
-    # TODO: works well only for small ledgers,
     ledger_data_dir = get_ledger_dir(args.node_name, args.client_name, args.network)
     read_copy_ledger_data_dir = None
     try:
-        read_copy_ledger_data_dir = make_copy_of_ledger(ledger_data_dir)
-        ledger = get_ledger(args.type, read_copy_ledger_data_dir)
+        # RocksDB supports real read-only mode and does not need to have a ledger copy.
+        if config.hashStore['type'].lower() != HS_ROCKSDB:
```
Comment (ashcherbakov) [inline scripts/read_ledger]: shouldn't it be `upper()`?

File: scripts/read_ledger [hunk 2]
```diff
@@ -153,15 +156,26 @@ def make_copy_of_ledger(data_dir):
...
+            # NOTE: such approach works well only for small ledgers.
+            tmp = make_copy_of_ledger(ledger_data_dir)
```
Comment (anikitinDSR) [inline scripts/read_ledger]: Maybe we should check, that shutil.copytree complete successfully? What if we have really big ledger, and will raised  "There is no space left on device" error?

Comment (sergey-shilov) [reply scripts/read_ledger]: Hmm, I think it's already done. If exception is risen then we go directly to finally block. The read_copy_ledger_data_dir variable is empty in this case, so we do nothing, but a user will see the error message.

**Comment by sergey-shilov** [comment]
Test this please.

**Pull request merged**